Track metadata changes

SamuABC opened the pull request:
resolves #179 

# Changes
- new boolean field in the mission table named "was_modified". If this field is true, the cli knows that new metadata for this mission needs to be saved in the filesystem
- restapi and fetchapi to access the was_modified field
- set was_modified to true when metadata in the frontend gets modified
- update sync command so that a mission only writes the metadata in the filesystem if was_modified is true for that mission
- a little bugfix where the tagPicker in DetailView did not fetch allTags and therefore was not able to add existing tags. This probably happened accidentally during refactoring

# Testing
I already tested it and it worked but for the reviewer:
You can try the following frontend actions and see if the database table entry for was_modified switches to true.
- change location
- change notes
- add new tag, modify a tag, add an existing tag, remove all tags, remove one tag

Or/and you can check if the sync command correctly only writes data into the filesystem if something was changed

DerSimi reviewed (changes requested):
Hi Samuel,
the current implementation in the frontend, checking for changes is error prone, due to the numerous cases one has to consider. It would be best, to include the new call directly in the Rest API functions. Furthermore, the test for the new fetch api function is missing. It does not contain much functionality, but still, the url should be tested.

The modification checking on notes and location change does not work as expected: For example, even when nothing is changed (open the dialogue and click instantly update), the modified entry is set. Consider including an additional check and compare the values from location with the one being set, if it differs, call the update Mission function. Same for notes.

The backend part works flawlessly, good job.

SamuABC commented:
Hi Simon, I forgot the fetch api tests. They are now added.
Also notes and location now only get changed when something was actually changed.
I moved the calls of the setWasModified function to the fetch api (I assume you meant the fetch api). This indeed is better for preventing errors.

pascal260303 commented:
When exporting the .drawio file to png you can get a better resolution by setting the zoom to over 100%. Previously I used 200%. Can you also do that?

DerSimi commented on frontend/app/utilities/TagList.tsx:
```diff
@@ -2,6 +2,7 @@ import { Badge, Group, Popover } from "@mantine/core";
 import { IconPencil } from "@tabler/icons-react";
 import { Tag } from "~/data";
 import { TagPicker } from "./TagPicker";
+import { setWasModified } from "~/fetchapi/missions";
```
```suggestion
```

DerSimi commented on frontend/app/pages/missions/Overview.tsx:
```diff
@@ -25,7 +25,7 @@ import { MissionData, RenderedMission, Tag } from "~/data";
 import { IconPencil } from "@tabler/icons-react";
 import { useNavigate } from "@remix-run/react";
-import { getMissions } from "~/fetchapi/missions";
+import { getMissions, setWasModified } from "~/fetchapi/missions";
```
```suggestion
import { getMissions } from "~/fetchapi/missions";
```

DerSimi reviewed (commented):
Hi Samuel, now I'm happy. Please look through my two suggestions, both remove unnecessary imports.

SamuABC commented:
> When exporting the .drawio file to png you can get a better resolution by setting the zoom to over 100%. Previously I used 200%. Can you also do that?

Yes, of course. I doubled the resolution
